Workflow failure: Release Build

## Workflow Failure

[Release Build](https://github.com/pulumi/pulumi-docker-containers/blob/main/.github/workflows/release.yml) has failed. See the list of failures below:

- [2024-03-28T14:02:09.000Z](https://github.com/pulumi/pulumi-docker-containers/actions/runs/8468277708)

- [2024-03-28T18:15:51.000Z](https://github.com/pulumi/pulumi-docker-containers/actions/runs/8468277708)

- [2024-03-28T23:32:41.000Z](https://github.com/pulumi/pulumi-docker-containers/actions/runs/8468277708)

- [2024-03-28T23:57:41.000Z](https://github.com/pulumi/pulumi-docker-containers/actions/runs/8468277708)

- [2024-03-29T00:27:25.000Z](https://github.com/pulumi/pulumi-docker-containers/actions/runs/8468277708)

- [2024-03-29T17:43:02.000Z](https://github.com/pulumi/pulumi-docker-containers/actions/runs/8468277708)

- [2024-03-29T18:20:04.000Z](https://github.com/pulumi/pulumi-docker-containers/actions/runs/8468277708)

- [2024-03-29T21:02:03.000Z](https://github.com/pulumi/pulumi-docker-containers/actions/runs/8468277708)

```
        npm ERR! code ECONNRESET
        npm ERR! errno ECONNRESET
        npm ERR! network request to https://registry.npmjs.org/get-caller-file/-/get-caller-file-2.0.5.tgz failed, reason: socket hang up
        npm ERR! network This is a problem related to network connectivity.
        npm ERR! network In most cases you are behind a proxy or have bad network settings.
        npm ERR! network 
        npm ERR! network If you are behind a proxy, please make sure that the
        npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
```

Re-running...

Hmm, got a different error on the latest run:

```
        npm ERR! code EPIPE
        npm ERR! syscall write
        npm ERR! errno EPIPE
        npm ERR! request to https://registry.npmjs.org/get-caller-file/-/get-caller-file-2.0.5.tgz failed, reason: write EPIPE
```

Network is unreliable installing dependencies for the test for Debian SDK images (nodejs, arm64):

```
        npm ERR! code ECONNRESET
        npm ERR! errno ECONNRESET
        npm ERR! network request to https://registry.npmjs.org/ansi-styles/-/ansi-styles-4.3.0.tgz failed, reason: Client network socket disconnected before secure TLS connection was established
        npm ERR! network This is a problem related to network connectivity.
        npm ERR! network In most cases you are behind a proxy or have bad network settings.
        npm ERR! network 
        npm ERR! network If you are behind a proxy, please make sure that the
        npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,10 @@
 # CHANGELOG
 
+## 3.112.0
+
+- Install the latest version of `npm` in the `pulumi/nodejs` image
+  ([#190](https://github.com/pulumi/pulumi-docker-containers/pull/190))
+
 ## 3.87.0
 
 - Upgrade Go to 1.21.1. ([#159](https://github.com/pulumi/pulumi-docker-containers/pull/159))

--- a/docker/nodejs/Dockerfile
+++ b/docker/nodejs/Dockerfile
@@ -24,7 +24,8 @@ RUN apt-get update -y && \
     apt-get install -y \
     curl \
     git \
-    ca-certificates
+    ca-certificates && \
+    npm install -g npm@10.5.1
 
 # Uses the workdir, copies from pulumi interim container
 COPY --from=builder /root/.pulumi/bin/pulumi /pulumi/bin/pulumi

--- a/docker/nodejs/Dockerfile.ubi
+++ b/docker/nodejs/Dockerfile.ubi
@@ -25,6 +25,7 @@ RUN microdnf install -y \
     tar \
     nodejs \
     ca-certificates && \
+    npm install -g npm@10.5.1 && \
     npm install -g yarn
 
 # Uses the workdir, copies from pulumi interim container

